great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -286,81 +366,121 @@ def _validate_batch_request(self, batch_request: RuntimeBatchRequest):
+            1. user has configured Assets at the RuntimeDataConnector-level (preferred).
+            2. user has not configured Assets and will pass in `data_asset_name` with the RuntimeBatchRequest.
 
-        1. refresh or create data_reference_cache
-        2. print batch_definition_count and example_data_references for each data_asset_names
-        3. also print unmatched data_references, and allow the user to modify the regex or glob configuration if necessary
+        In the case of #1, the get_available_data_asset_names() will return the list of configured Assets and base
+        self_check() will be called.
 
-        However, in the case of the RuntimeDataConnector there is no example data_asset_names until the data is passed
-        in through the RuntimeBatchRequest. Therefore, there will be a note displayed to the user saying that
-        RuntimeDataConnector will not have data_asset_names until they are passed in through RuntimeBatchRequest.
+        In the case of #2  there are no example data_asset_names until the data is passed in through the
+        RuntimeBatchRequest. Therefore, there will be a note displayed to the user saying that RuntimeDataConnector
+        will not have data_asset_names until they are passed in through RuntimeBatchRequest.
 
         Args:
             pretty_print (bool): should the output be printed?
             max_examples (int): how many data_references should be printed?
-
         Returns:
             report_obj (dict): dictionary containing self_check output
         """
+        if len(self._data_references_cache) == 0:
+            self._refresh_data_references_cache()
+
         if pretty_print:
             print(f"\t{self.name}:{self.__class__.__name__}\n")
         asset_names = self.get_available_data_asset_names()
-        asset_names.sort()
         len_asset_names = len(asset_names)
 
-        report_obj = {
-            "class_name": self.__class__.__name__,
-            "data_asset_count": len_asset_names,
-            "example_data_asset_names": asset_names[:max_examples],
-            "data_assets": {},
-            "note": "RuntimeDataConnector will not have data_asset_names until they are passed in through RuntimeBatchRequest"
-            # "data_reference_count": self.
-        }
-        if pretty_print:
-            print(
-                f"\tAvailable data_asset_names ({min(len_asset_names, max_examples)} of {len_asset_names}):"
-            )
-            print(
-                "\t\t"
-                + "Note : RuntimeDataConnector will not have data_asset_names until they are passed in through RuntimeBatchRequest"
-            )
-
-        unmatched_data_references = self.get_unmatched_data_references()
-        len_unmatched_data_references = len(unmatched_data_references)
-
-        if pretty_print:
+        if len_asset_names > 0:
+            return super().self_check()
+        else:
+            report_obj = {
```

Comment:
@Shinnnyshinshin type hints throughout, please, including that for `report_obj`.  Thanks!

Reply:
done!